### ISSUE TITLE
Hyperparameters explanation

### ISSUE THREAD

**Issue by simalessandri** [ultralytics/yolov5#13234]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi, i searched in the doc but I don't find any information more than the commend part after each hyperparameter.. in particular i'm interestend in the following

> fl_gamma: 0.0 # focal loss gamma (efficientDet default gamma=1.5)
> hsv_h: 0.015 # image HSV-Hue augmentation (fraction)
> hsv_s: 0.7 # image HSV-Saturation augmentation (fraction)
> hsv_v: 0.4 # image HSV-Value augmentation (fraction)
> degrees: 10 # image rotation (+/- deg)
> translate: 0.1 # image translation (+/- fraction)
> scale: 0.9 # image scale (+/- gain)
> shear: 0.00 # image shear (+/- deg)
> perspective: 0.0 # image perspective (+/- fraction), range 0-0.001)

There are some hyp that refer to probability, so you define the probability that during training that kind of action is performed, but what about these? For example if i set degrees to 10 does it mean that all images during training are rotated by 10? Or that it will randomly rotated by +- 10?

Thanks for the clarification

**Comment by glenn-jocher** [comment]
@simalessandri hi there,

Thank you for your question and for searching the documentation and issues beforehand! Let's dive into the hyperparameters you've mentioned:

1. **fl_gamma**: This parameter is used for Focal Loss, which is designed to address class imbalance by down-weighting easy examples and focusing on hard ones. A value of `0.0` means Focal Loss is not used, while a higher value (e.g., 1.5) increases the focus on hard examples.

2. **hsv_h, hsv_s, hsv_v**: These parameters control the augmentation of images in the HSV color space:
   - `hsv_h`: Adjusts the hue of the image. A value of `0.015` means the hue can be adjusted by ±1.5%.
   - `hsv_s`: Adjusts the saturation. A value of `0.7` means the saturation can be adjusted by ±70%.
   - `hsv_v`: Adjusts the value (brightness). A value of `0.4` means the brightness can be adjusted by ±40%.

3. **degrees**: This parameter controls image rotation. A value of `10` means that during training, images can be randomly rotated by up to ±10 degrees.

4. **translate**: This parameter controls image translation. A value of `0.1` means that images can be randomly translated by up to ±10% of the image size in both x and y directions.

5. **scale**: This parameter controls image scaling. A value of `0.9` means that images can be randomly scaled by a factor between `1 - 0.9` and `1 + 0.9`.

6. **shear**: This parameter controls image shearing. A value of `0.0` means no shearing is applied. If set to a value like `10`, images can be randomly sheared by up to ±10 degrees.

7. **perspective**: This parameter controls image perspective transformation. A value of `0.0` means no perspective transformation is applied. If set to a value like `0.001`, images can be randomly transformed within this range.

To summarize, these hyperparameters define the range within which random augmentations are applied to the images during training. For example, setting `degrees` to `10` means that each image can be randomly rotated by any angle between -10 and +10 degrees, not that every image will be rotated by exactly 10 degrees.

I hope this clarifies your questions! If you have any more, feel free to ask. 😊

**Comment by simalessandri** [comment]
Your explanation was incredibly clear! Thank you so much!

**Comment by glenn-jocher** [comment]
@simalessandri thank you for the kind words! 😊 I'm glad the explanation was helpful to you. If you have any more questions or need further clarification on any other aspects of YOLOv5, feel free to ask. The YOLO community and the Ultralytics team are always here to help!

Happy training! 🚀
